### PR TITLE
Run the upstream test suite in CI (fixes #8)

### LINKED ISSUE #8: Run the upstream testsuite against our code in the CI
Upstream tests are here:
https://git.savannah.gnu.org/cgit/diffutils.git/tree/tests

### PR DESCRIPTION
**Pull request by oSoMoN**
The script takes a shortcut to fetch only the test suite from the upstream repository and carefully avoids running the autotools machinery which is time-consuming and resource-intensive, and doesn't offer the option to not build the upstream binaries.
As a consequence, the environment in which the tests are run might not match exactly that used when the upstream tests are run through the autotools. Whether tests are failing because of an incorrect environment remains to be investigated.

Unsurprisingly a number of tests are failing because of missing options in the current implementation or discrepancy in the output. These will need to be implemented/fixed.

### CODE REVIEW

**Comment by sylvestre** [comment]
The job should not fail.
And we should generate a json at the end like the coreutils but it can be done in another pr.

Please let me know when it is ready for review

**Comment by oSoMoN** [comment]
> The job should not fail. And we should generate a json at the end like the coreutils but it can be done in another pr.

I tweaked the workflow to not fail when run in CI. I'll look into generating that JSON data as soon as I get a chance.


> Please let me know when it is ready for review

Almost there, I would like to allow `run-upstream-testsuite.sh` to run selected tests only, and to report results more verbosely to ease investigation of failures. I'm converting this PR to draft until it's ready for review.

**Comment by sylvestre** [comment]
Do you think it would be possible to show the errors in the log ? thanks

**Comment by oSoMoN** [comment]
I believe this is now ready for review.

> Do you think it would be possible to show the errors in the log ? thanks

I added a script that prints out details about every test (including the contents of stdout and stderr) in a markdown format, convenient for filing issues to track tests with an unexpected result. When rendered it looks like this:

# test: excess-slash
result: FAIL
url: https://git.savannah.gnu.org/cgit/diffutils.git/tree/tests/excess-slash?id=6c88db1f7bfc25b1480d9876dac25024aa688544
## stdout
```
diff -u /dev/null err
--- /dev/null	1970-01-01
+++ err	1970-01-01
+Error: "Unknown option: r"
--- expected-out	
+++ out	
@@ -1 +0,0 @@
-Only in b/f: g
Not removing temporary directory /tmp/tmp.3jNgxNWI94/diffutils/tests/gt-excess-slash.YClC
```

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -49,6 +49,23 @@ jobs:
       - run: rustup component add clippy
       - run: cargo clippy -- -D warnings
 
+  upstream-testsuite:
+    name: Upstream test suite
```
Comment (sylvestre) [inline .github/workflows/ci.yml:53]:
```suggestion
  gnu-testsuite:
    name: GNU test suite
```

**Comment by sylvestre** [comment]
excellent, thanks

**Pull request merged**